index.html

```diff
@@ -34,14 +34,21 @@
 
       const ldclient = LDClient.initialize(clientSideID, user);
 
-      function render() {
-        const flagValue = ldclient.variation(flagKey, false);
-        const label = 'Feature flag ' + flagKey + ' is ' + flagValue + ' for this user';
-        div.replaceChild(document.createTextNode(label), div.firstChild);
-      }
+      ldclient.waitForInitialization().then(() => {
+        div.replaceChild(document.createTextNode('SDK successfully initialized!'), div.firstChild);
+
+        function render() {
+          const flagValue = ldclient.variation(flagKey, false);
+          const label = 'Feature flag ' + flagKey + ' is ' + flagValue + ' for this user';
+          div.replaceChild(document.createTextNode(label), div.firstChild);
+        }
+
+        render();
```

Comment:
Previously, this was `ldclient.on('ready', render);`. However, once we're in the `waitForInitialization` block, I had to call `render()` explicitly the first time to get the "Feature flag xx is true/false for this user" message to appear. So I'm not sure that this is correct.